### ISSUE TITLE
Dataset bundles for easy navigation in histories

### ISSUE THREAD

**Issue by nekrut** [galaxyproject/galaxy#2450]

**Comment by nekrut** [comment]
If a history has more than six datasets representing two or more parallel analyses, it becomes a hassle to untangle (imagine if there are 100s). One potential solution is to allow users to group datasets using tool invocation chains into bundles as shown below. Such bundles should be expandable and collapsable:

![screen shot 2016-06-01 at 9 49 09 am](https://cloud.githubusercontent.com/assets/4291636/15780704/f629dfce-2970-11e6-861f-a011bcbddcac.png)

The bottom line: we should be able to view history by tool invocation chains. Here is an example: 
- I have two datasets
- I ran a workflow with 40 steps
- Now I have 82 datasets in history
- Now I would like to concatenate results of two workflow runs
- With "bundled" view I will immediately be able to see terminal datasets produced by workflows and will be able to drag them into the tool form.

**Comment by peterjc** [comment]
The very similar idea of visually grouping of all the outputs from a single workflow came up today. This is slightly simpler than the example above - it would be automatic (not a user-editable bundling), not include the inputs, and the grouping doesn't need to change the order as each workflow's output files will all be consecutive.

**Comment by hexylena** [comment]
Could this be accomplished sufficiently with collections and nametags that we should close it?
